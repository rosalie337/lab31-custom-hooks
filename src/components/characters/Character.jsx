import React from 'react'
import PropTypes from 'prop-types'

function Characters({name, url}) {
    return (
        <>
            <figure>
                <img src={url} />
                <figcaption>{name}</figcaption>  
            </figure>
        </>
    )
}

Characters.propTypes = {
    name: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,

}

export default Characters

