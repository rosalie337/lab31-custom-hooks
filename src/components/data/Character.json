[
    {
      "_id": "5da237699734fdcb7bef8f56",
      "name": "Miles Shortman",
      "image": "https://vignette.wikia.nocookie.net/heyarnold/images/2/2e/Miles.jpg/revision/latest/scale-to-width-down/243?cb=20101115140639"
    }
]