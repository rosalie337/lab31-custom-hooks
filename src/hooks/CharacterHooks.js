import { useState, useEffect} from 'react';
import { getCharacterDetail, getCharacters } from '../services/fetchApi'
import loading from '../loading/Loading';

export const useCharacters = () => {
    const [loading, setLoading] = useState(true);
    const [characters, setCharacters] = useState([]);

    useEffect(() => {
        getCharacters()
          .then(newCharacters => {
            setCharacters(newCharacters);
            setLoading(false);
          });
      }, []);

    return {
        loading,
        characters
    };
};

export const useCharacterDetails = (id) => {
    const [character, setCharacter] = useState({});
    const [loading, setLoading] = useState(true);
  
    useEffect(() => {
      getCharacterDetail(id)
        .then(newCharacter => {
          setCharacter(newCharacter);
          setLoading(false);
        });
    }, []);

    return {
        loading,
        character
    };
}