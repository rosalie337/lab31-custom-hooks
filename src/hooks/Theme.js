/* eslint-disable no-unused-vars */
import React, { useState, createContext, useContext } from 'react';

export const CharacterTheme = createContext(null);

export const NewTheme = ({ children }) => {
    const [theme, setTheme] = useState('dark');
    
    const toggle = () => {
        if (theme === 'dark')setTheme('light');
        else setTheme('dark')
    }
    return (
        <CharacterTheme.Provider value = {{ theme, toggle }}>
            {children}
        </CharacterTheme.Provider>
    );
};


export const useTheme = () => {
    const { theme, toggle } = useContext(CharacterTheme);
    return {
        theme,
        toggle
    }
}

