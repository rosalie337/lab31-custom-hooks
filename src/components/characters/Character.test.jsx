/* eslint-disable max-len */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Character from './Character';
import { NewTheme } from '../../hooks/Theme';

describe('Character component', () => {
  afterEach(() => cleanup());
  it('renders Character', () => {
    const { asFragment } = render(<NewTheme><Character
      name="Arik"
      url="https://vignette.wikia.nocookie.net/avatar/images/6/6f/Arik.png/revision/latest?cb=20140808184002"
    /></NewTheme>);    
      
    expect(asFragment()).toMatchSnapshot();
  });
});