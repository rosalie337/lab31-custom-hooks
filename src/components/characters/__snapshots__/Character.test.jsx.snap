// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<DocumentFragment>
  <div>
    <figure>
      <img
        src="https://vignette.wikia.nocookie.net/avatar/images/6/6f/Arik.png/revision/latest?cb=20140808184002"
      />
      <figcaption>
        Arik
      </figcaption>
    </figure>
  </div>
</DocumentFragment>
`;
